### PR TITLE
 Attack Calc - Display warning about max possible DCR

### PR DESCRIPTION
**Pull request by Ekeh**
This PR checks the total amount of DCR needed for the calculated PoS attack and displays a warning indicator when the amount of DCR needed for the attack is greater than the total amount of DCR in circulation.
attackcost1
attackcost2

### CODE REVIEW

**Comment by raedah** [comment]
Tested, and works as expected.

File: public/js/controllers/attackcost_controller.js [hunk 1]
```diff
@@ -122,7 +122,8 @@ export default class extends Controller {
       'total', 'totalDCRPos', 'totalDeviceCost', 'totalElectricity', 'totalExtraCostRate', 'totalKwh',
       'totalPos', 'totalPow', 'graph', 'labels', 'projectedTicketPrice', 'projectedTicketPriceIncrease', 'attackType',
       'attackPosPercentAmountLabel', 'dcrPriceLabel', 'totalDCRPosLabel',
-      'projectedPriceDiv'
+      'projectedPriceDiv', 'attackNotPossibleWrapperDiv', 'coinSupply',
```
Comment (ademuanthony) [inline public/js/controllers/attackcost_controller.js:125]: remove the unused target `coinSupply`

File: public/js/controllers/attackcost_controller.js [hunk 2]
```diff
@@ -469,4 +471,16 @@ export default class extends Controller {
   showAll (targets) {
     targets.forEach(el => el.classList.remove('d-none'))
   }
+
+  calcPosWarning (DCRNeed) {
```
Comment (ademuanthony) [inline public/js/controllers/attackcost_controller.js:475]: What about `showPosCostWarning`

File: public/js/controllers/attackcost_controller.js (hunk 1 above)
Comment (Ekeh) [reply public/js/controllers/attackcost_controller.js:125]: coinSupply target is used. Please check line 476.

File: public/js/controllers/attackcost_controller.js [hunk 3]
```diff
@@ -113,16 +113,14 @@ export default class extends Controller {
...
-      'externalAttackPosText', 'additionalDcr', 'newTicketPoolValue', 'internalAttackPosText',
-      'additionalHashRate', 'newHashRate', 'targetPos', 'targetPow',
-      'ticketAttackSize', 'ticketPoolAttack', 'ticketPoolSize', 'ticketPoolSizeLabel',
-      'ticketPoolValue', 'ticketPrice', 'tickets', 'ticketSizeAttack', 'durationLongDesc',
-      'total', 'totalDCRPos', 'totalDeviceCost', 'totalElectricity', 'totalExtraCostRate', 'totalKwh',
-      'totalPos', 'totalPow', 'graph', 'labels', 'projectedTicketPrice', 'projectedTicketPriceIncrease', 'attackType',
-      'attackPosPercentAmountLabel', 'dcrPriceLabel', 'totalDCRPosLabel',
-      'projectedPriceDiv'
+      'deviceCost', 'deviceDesc', 'deviceName', 'external', 'internal', 'internalHash', 'kwhRate', 
+      'kwhRateLabel', 'otherCosts', 'otherCostsValue', 'priceDCR', 'internalAttackText', 'targetHashRate',
```
Comment (chappjc) [inline public/js/controllers/attackcost_controller.js:117]: The trailing spaces on 116 and 117 are breaking the webpack build:

```
/home/jon/github/decred/dcrdata/public/js/controllers/attackcost_controller.js
  116:99   error  Trailing spaces not allowed  no-trailing-spaces
  117:107  error  Trailing spaces not allowed  no-trailing-spaces

✖ 2 problems (2 errors, 0 warnings)
  2 errors and 0 warnings potentially fixable with the `--fix` option.

 @ ./public/js/controllers sync \.js$ ./attackcost_controller.js
 @ ./public/index.js
```

```suggestion
      'deviceCost', 'deviceDesc', 'deviceName', 'external', 'internal', 'internalHash', 'kwhRate',
      'kwhRateLabel', 'otherCosts', 'otherCostsValue', 'priceDCR', 'internalAttackText', 'targetHashRate',
```

File: views/attackcost.tmpl [hunk 4]
```diff
@@ -348,47 +349,47 @@
...
-                <span> * </span>
-                <span data-target="attackcost.dcrPriceLabel">0</span><span class="fs11"> USD/DCR</span>
-              </span>)
-            </div>
-
-          </div>
-
+        {{- /* CALCULATED PoS WARNING DISPLAY */ -}}
+        <div data-target="attackcost.attackNotPossibleWrapperDiv" class="mt-1 ml-2 blink-container d-none" style="color: #f12222; margin-left: 8px; ">
+            Attack not possible. Total coin supply is <span  data-target="attackcost.coinSupply" class="font-weight-bold">0</span> <span class="fs11">DCR</span>.
```
Comment (chappjc) [inline views/attackcost.tmpl:358]:
```suggestion
            Attack not possible. Total coin supply is <span data-target="attackcost.coinSupply" class="font-weight-bold">0</span> <span class="fs11">DCR</span>.
```

File: views/attackcost.tmpl [hunk 5]
```diff
@@ -348,47 +349,47 @@
...
         Total attack cost:
-        <span class="position-relative text-secondary font-weight-bold">
+        <span class="position-relative text-secondary font-weight-bold" data-target="attackcost.totalAttackCostContainer">
           $<span class="font-weight-bold" data-target="attackcost.total">0</span>
           <span class="fs11">USD</span>
         </span>
       </div>
     </div>
-    </div>
+    </div>
```
Comment (chappjc) [inline views/attackcost.tmpl:392]:
```suggestion
    </div>
```

**Pull request merged**